cli/src/semgrep/target_manager.py

```diff
@@ -339,7 +339,12 @@ def validate_path(self, _: Any, value: Path) -> None:
 
     def _is_valid_file_or_dir(self, path: Path) -> bool:
         """Check this is a valid file or directory for semgrep scanning."""
-        return os.access(str(path), os.R_OK) and not path.is_symlink()
+        if not os.path.exists(str(path)):
+            return False
+        stat_results = os.stat(str(path))
```

Comment:
```suggestion
        stat_results = path.stat()
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat